adam-core/src/main/scala/org/bdgenomics/adam/converters/AlignmentRecordConverter.scala

```diff
@@ -42,13 +47,8 @@ class AlignmentRecordConverter extends Serializable {
     val readNameSuffix =
       if (maybeAddSuffix &&
         !AlignmentRecordConverter.readNameHasPairedSuffix(adamRecord) &&
-        adamRecord.getFirstOfPair != null) {
-        "/" + (
-          if (adamRecord.getFirstOfPair)
-            "1"
-          else
-            "2"
-        )
+        adamRecord.getReadPaired) {
+        "/%d".format(adamRecord.getReadNum + 1)
       } else {
         ""
       }
```

Comment:
What the advantage of having the discussion on #818 instead of in this PR? Just curious.